_posts/2021-11-18-keras-updates/keras-updates.Rmd

```diff
@@ -0,0 +1,159 @@
+---
+title: "Keras for R is back!"
+description: > 
+  For a while, it may have seemed that Keras for R were in some undecidable state, like Schrödinger's cat before inspection. It is high time to  correct that impression. Keras for R is back, with two recent releases adding powerful capabilities that considerably lighten previously tedious tasks. 
+  This post provides a high-level overview. Future posts will go into more detail on some of the most helpful new features, as well as dive into the powerful low-level enhancements that make the former possible.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+  - name: Tomasz Kalinowski
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanakalinowskikeras
+date: 11-18-2021
+categories:
+  - Packages/Releases
+  - TensorFlow/Keras
+  - R
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/preview.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Before we even talk about new features, let us answer the obvious question. Yes, there will be a second edition of *Deep Learning for R*! Reflecting what has been going on in the meantime, the new edition covers an extended set of proven architectures; at the same time, you'll find that intermediate-to-advanced designs already present in the first edition have become rather more intuitive to implement, thanks to the new low-level enhancements alluded to in the summary.
+
+But don't get us wrong -- the scope of the book is completely unchanged. It still is the perfect choice for people new to machine learning and deep learning. Starting from the basic ideas, it systematically progresses to intermediate and advanced topics, leaving you with both conceptual understanding and a bag of useful application templates.
+
+*Now, what has been going on with Keras?*
+
+## State of the ecosystem
+
+Let us start with a characterization of the ecosystem, and a few words on history.
+
+In this post, when we say *Keras*, we mean R -- as opposed to Python -- *Keras*[^1]. Now, this immediately translates to the R package `keras`. But `keras` alone wouldn't get you far. While `keras` provides the high-level functionality -- neural network layers, optimizers, workflow management, and more -- the basic data structure operated upon, *tensors*, lives in `tensorflow`. Thirdly, as soon as you'll need to perform less-then-trivial pre-processing, or can no longer keep the whole training set in memory because of its size, you'll want to look into `tfdatasets`.
```

Comment:
```suggestion
In this post, when we say *Keras*, we mean R -- as opposed to Python -- *Keras*[^1]. Now, this immediately translates to the R package `keras`. But `keras` alone wouldn't get you far. While `keras` provides the high-level functionality -- neural network layers, optimizers, workflow management, and more -- the basic data structures operated upon, *tensors*, live in `tensorflow`. Thirdly, as soon as you need to perform less-then-trivial pre-processing, or can no longer keep the whole training set in memory because of its size, you'll want to look into `tfdatasets`.
```

Reply:
Since tensors is plural, it makes sense to speak of the plural in general

Reply:
honestly i'd like to leave it singular - as in, "tensors constitute the basic data structure"

Reply:
Go for it.